build/parser.mjs

```diff
@@ -941,12 +941,20 @@ class Parser {
    * Adds releaseDate, vaultDate and estimatedVaultDate to all primes using
    * data from "Ducats or Plat".
    * @param {Item} item data to append vault data to
+   * @param {module:warframe-items.Category} category of the data
+   * @param {WikiaData} wikiaData from wikia to apply
    * @param {VaultData} vaultData to look up data for the {@param item}
    */
-  addVaultData(item, vaultData) {
+  addVaultData(item, category, wikiaData, vaultData) {
+    if (!['weapons', 'warframes', 'mods', 'upgrades'].includes(category.toLowerCase())) return;
     if (!item.name.endsWith('Prime')) return;
+
     const target = vaultData.find((i) => i.Name.toLowerCase() === item.name.toLowerCase());
 
+    if (item.type === 'Archwing') category = 'archwings';
+    if (category === 'Upgrades') category = 'mods';
```

Comment:
reassignment of function parameters is not very good by design, in this case nothing bad should happen, but could you give another name to category in this context like vaultCategory? And also if you're changing function parameters to use in the rest of the function do it at the first chance you have, it'll be easier to read.

For example
```js
let vaultCategory = category;
if (item.type === 'Archwing') vaultCategory = 'archwings';
if (vaultCategory === 'Upgrades') vaultCategory = 'mods';

if (!['weapons', 'warframes', 'mods', 'upgrades'].includes(vaultCategory.toLowerCase())) return;
if (!item.name.endsWith('Prime')) return;

const target = vaultData.find((i) => i.Name.toLowerCase() === item.name.toLowerCase());
```

Reply:
This is actually not allowed in dart either so not sure why I did it here 🤣